services/streammanager/googlesheets/googlesheetsmanager.go

```diff
@@ -161,46 +159,93 @@ func insertDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, sprea
 		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
 
-	if isHeader {
-		_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	_, err = sheetsClient.Spreadsheets.Values.Update(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+
+	return err
+}
+
+// Wrapper func to append row data list,
+// Returns error for failure cases of API calls otherwise returns nil
+func insertRowDataToSheet(sheetsClient *sheets.Service, spreadSheetId string, spreadSheetTab string, dataList [][]interface{}) error {
+	// Creating value range for inserting row into sheet
+	var vr sheets.ValueRange
+	vr.MajorDimension = "ROWS"
+	vr.Range = spreadSheetTab + "!A1"
+	vr.Values = dataList
+	var err error
 
-	} else {
-		_, err = sheetsClient.Spreadsheets.Values.Append(spreadSheetId, spreadSheetTab+"!A1", &vr).ValueInputOption("RAW").Do()
+	if sheetsClient == nil {
+		return fmt.Errorf("[GoogleSheets] error  :: Failed to initialize google-sheets client")
 	}
```

Comment:
Can we move this validation (and from the header method) to the caller function?

Reply:
Moved to caller function